### PR TITLE
Auto-create config dir if not exists

### PR DESCRIPTION
**Pull request by requilence**
For example It was a fatal error when trying to run the cagent on linux system without `/etc/cagent` directory presented

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -212,9 +212,17 @@ func GenerateDefaultConfigFile(mvc *MinValuableConfig, configFilePath string) er
...
+		err := os.MkdirAll(configPathDir, os.ModePerm)
+		if err != nil {
+			return fmt.Errorf("failed to auto-create the default Config file directory '%s': %s", configPathDir, err.Error())
+		}
+	}
+
 	var f *os.File
 	if f, err = os.OpenFile(configFilePath, os.O_WRONLY|os.O_CREATE, 0666); err != nil {
-		return fmt.Errorf("failed to create the default Config file: '%s'", configFilePath)
+		return fmt.Errorf("failed to create the default Config file at '%s': %s", configFilePath, err.Error())
```
Comment (soupdiver) [inline config.go]: just a small note, why is everything non-capital _but_ `Config` was capitalised?

Comment (requilence) [reply config.go]: it was made this way by @troian. But yeah, its better to make it lowercase

Comment (soupdiver) [reply config.go]: Are we going to update this?

Comment (requilence) [reply config.go]: fixed

**Pull request merged**